src/DependencyInjection/Compiler/AddDependencyCallsCompilerPass.php

```diff
@@ -382,11 +396,19 @@ public function fixTemplates(ContainerBuilder $container, Definition $definition
 
         $definedTemplates = $overwrittenTemplates['view'] + $definedTemplates;
 
-        if ($container->getParameter('sonata.admin.configuration.templates') !== $definedTemplates) {
-            $definition->addMethodCall('setTemplates', [$definedTemplates]);
+        $templateRegistryId = $serviceId.'.template_registry';
+        $templateRegistryDefinition = $container
+            ->register($templateRegistryId, TemplateRegistry::class)
+            ->addTag('sonata.admin.template_registry')
+            ->setPublic(true); // Temporary fix until we can support service locators
+
+        if ($globalTemplates !== $definedTemplates) {
```

Comment:
Now that I see this code..

if `globalTemplates` is different from `definedTemplates` you use `definedTemplates`, but if they are equal, you use `globalTemplates`...

Doesn't that mean that you always use definedTemplates? Or am I missing something?

Reply:
Yeah, I'm starting to get more and more confused every time I look at this code. The original code uses the parameter by name instead of the actual array if the templates do not differ.
The PR you mentioned changes that to getting the value of that parameter instead of using the name.

In the end, it looks like that entire if-statement is not needed.

Reply:
Running the code without the if-statement and without any `$globalTemplates`, the end result is exactly the same.

Reply:
https://github.com/sonata-project/SonataAdminBundle/pull/4580

This was done here... Maybe it has a good reason, but I don't know then why it does not work according to #4842 

Can you help us with this @nicolas-grekas ?

Reply:
Seems like it is related to: https://github.com/symfony/symfony/pull/23143 

But that's only for SF 3.4, maybe older versions can't resolve that parameters.

Reply:
Looking at the generated container shws the difference (at least at #4580 times): a reference to the parameter is dumped instead of its value. That was the goal.